float32 xDist
float32 yDist
float32 eucliDist